strax/processing/general.py

```diff
@@ -408,34 +408,34 @@ def touching_windows(things, containers, window=0):
 @numba.njit(nogil=True, cache=True)
 def _touching_windows(thing_start, thing_end,
                       container_start, container_end,
-                      window=0):
-    result = np.zeros((len(container_start), 2), dtype=np.int32)
+                      window=0, endtime_sort_kind='mergesort'):
     n = len(thing_start)
+    thing_end_argsort = np.argsort(thing_end, kind=endtime_sort_kind)
+    thing_end_sort = thing_end[thing_end_argsort]
+    container_end_argsort = np.argsort(container_end, kind=endtime_sort_kind)
```

Comment:
Why can we not use the default sorting algorithm here?

Reply:
To make the sorting stable. We want to return the first index of `things` which is overlapping with `containers`.